ipcore/avalon_config.py

```diff
@@ -0,0 +1,70 @@
+import json
+from pathlib import Path
+
+from .util import Register, Audio, DataType
+
+class AvalonConfig:
+    """Describe configuration to generate avalon wrapper."""
+
+    def __init__(self, target, audio_in, audio_out, entity_name, registers=None, working_dir="", is_sample_based=False):
+        """Initialize an empty configuration."""
+        self.registers = registers or []
+        self.target_system = target
+        self.working_dir = working_dir
+        self.audio_in = audio_in
+        self.audio_out = audio_out
+        self.entity_name = entity_name
+        self.is_sample_based = is_sample_based
+
+    @staticmethod
+    def parse_json(config_filepath):
+        """Parse an AvalonConfig from a JSON file.
+
+        Parameters
+        ----------
+        config_filepath : str
+            file path to JSON file to parse
+
+        Returns
+        -------
+        AvalonConfig
+            Parsed configuration
+        """
+        with open(config_filepath, "r") as file:
+            in_str = file.read()
+        modeljson = json.loads(in_str)
+
+        working_dir = Path(config_filepath).parent.name
+        target = modeljson["system"]["target"]
+        json_audio_in = modeljson["system"]["audioIn"]
+        audio_in = Audio(
+            DataType(
+                json_audio_in["wordLength"],
+                json_audio_in["fractionLength"],
+                json_audio_in["signed"]),
+            json_audio_in["numberOfChannels"]
+        )
```

Comment:
The indentation is quite intentional there, although I am curious to see if autopep8 does anything different. Because I am making an object inside of another object, I have the second constructor's inputs indented more than the first.

Reply:
Ah, okay. I missed that there were two constructors. Perhaps having the `)` of the first constructor on the next line would clear things up a bit. Either way, that makes perfect sense. I too am curious to see if autopep8 does anything different.